tap_google_ads/streams.py

```diff
@@ -750,92 +795,33 @@ def sync(self, sdk_client, customer, stream, config, state, query_limit):
-                transformed_message = self.transform_keys(json_message)
-                record = transformer.transform(transformed_message, stream["schema"])
-                record["_sdc_record_hash"] = generate_hash(record, stream_mdata)
-                record = self.add_account_name(customer, record)
-
-                singer.write_record(stream_name, record)
-
-        new_bookmark_value = {replication_key: utils.strftime(query_date)}
-        singer.write_bookmark(state, stream["tap_stream_id"], customer["customerId"], new_bookmark_value)
-
-        singer.write_state(state)
-
-
-class OneDayResultsReportStream(ReportStream):
-    def sync(self, sdk_client, customer, stream, config, state, query_limit):
-        gas = sdk_client.get_service("GoogleAdsService", version=API_VERSION)
-        resource_name = self.google_ads_resource_names[0]
-        stream_name = stream["stream"]
-        stream_mdata = stream["metadata"]
-        selected_fields = get_selected_fields(stream_mdata)
-        replication_key = "date"
-        state = singer.set_currently_syncing(state, [stream_name, customer["customerId"]])
-        singer.write_state(state)
-
-        conversion_window = timedelta(
-            days=get_conversion_window(config)
-        )
-        conversion_window_date = utils.now().replace(hour=0, minute=0, second=0, microsecond=0) - conversion_window
-
-        bookmark_object = singer.get_bookmark(state, stream["tap_stream_id"], customer["customerId"], default={})
-
-        bookmark_value = bookmark_object.get(replication_key)
-
-        query_date = get_query_date(
-            start_date=config["start_date"],
-            bookmark=bookmark_value,
-            conversion_window_date=singer.utils.strftime(conversion_window_date)
-        )
-
-        end_date = config.get("end_date")
-        if end_date:
-            end_date = utils.strptime_to_utc(end_date)
+        if stream_name in REPORTS_ONE_DAY_ONLY:
+            LOGGER.info(f"Stream: {stream_name} supports quering by one day only. Setting split_by_period value to 1 day.")
+            split_by_period = relativedelta(days=1)
         else:
-            end_date = utils.now()
-
-        if stream_name in REPORTS_WITH_90_DAY_MAX:
-            cutoff = end_date.replace(hour=0, minute=0, second=0, microsecond=0) - timedelta(days=90)
-            query_date = max(query_date, cutoff)
-            if query_date == cutoff:
-                LOGGER.info(
-                    f"Stream: {stream_name} supports only 90 days of data. "
-                    f"Setting start date to {utils.strftime(query_date, '%Y-%m-%d')}."
-                )
+            split_by_period = get_split_by_period(config)
 
         while query_date <= end_date:
-            query = create_one_day_report_query(resource_name, selected_fields, query_date)
-            LOGGER.info(f"Requesting {stream_name} data for {utils.strftime(query_date, '%Y-%m-%d')}.")
+            if split_by_period is None:
```

Comment:
ideally we would keep the code that generates date range to report on separate, similar to what we do internally.

Reply:
Agree. Will do if testing uncovers any defects.